Generate serialization of WebCore::Length

achristensen07 opened the pull request:
#### 60d33b0e31c939b2fc947aeabde3137782d89f6c
<pre>
Generate serialization of WebCore::Length
<a href="https://bugs.webkit.org/show_bug.cgi?id=265600">https://bugs.webkit.org/show_bug.cgi?id=265600</a>
<a href="https://rdar.apple.com/119000026">rdar://119000026</a>

Reviewed by Brady Eidson.

Changing the data storage of WebCore::Length to a std::variant would increase sizeof(Length)
because of padding, but I achieved the same goal for serialization by making a variant accessor
and a constructor that takes a variant.

* Source/WebCore/platform/Length.cpp:
(WebCore::Length::typeFromIndex):
(WebCore::Length::Length):
(WebCore::Length::ipcData const):
(WebCore::Length::floatOrInt const):
* Source/WebCore/platform/Length.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder&lt;Length&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;Length&gt;::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in.orig: Copied from Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in.

Canonical link: <a href="https://commits.webkit.org/271620@main">https://commits.webkit.org/271620@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6c6da923b9d74187b7231d0b025dc1eee7b38606

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29006 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/7679 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/30363 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/31621 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/26430 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/29598 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/9811 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/4994 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/26453 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/29278 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/6339 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/24887 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5515 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/5639 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/25913 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/32968 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26507 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/26350 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/31892 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5616 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3792 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/29678 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7272 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6113 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3734 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6137 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

beidson reviewed (approved):
LGTM once bots are happy

achristensen07 committed:
Generate serialization of WebCore::Length
https://bugs.webkit.org/show_bug.cgi?id=265600
rdar://119000026

Reviewed by Brady Eidson.

Changing the data storage of WebCore::Length to a std::variant would increase sizeof(Length)
because of padding, but I achieved the same goal for serialization by making a variant accessor
and a constructor that takes a variant.

* Source/WebCore/platform/Length.cpp:
(WebCore::Length::typeFromIndex):
(WebCore::Length::Length):
(WebCore::Length::ipcData const):
(WebCore::Length::floatOrInt const):
* Source/WebCore/platform/Length.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder<Length>::encode): Deleted.
(IPC::ArgumentCoder<Length>::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in.orig: Copied from Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in.

Canonical link: https://commits.webkit.org/271620@main
